### DISCUSSION TITLE
Public IP check plugin

### DISCUSSION THREAD

**Discussion by mihakralj** [mimugmail/opn-repo#28] [Ideas]
Looking for a feedback on the plugin that I am putting together.

IPcheck pulls data from external API providers, self-verifying the exposure of WAN port:
- https://api.ipify.org?format=json - returns the visible IPv4 address
- https://api64.ipify.org?format=json - returns visible IPv6 address (if exists)
- https://vpnapi.io/api/<ip> - returns basic verification: location, network, ISP, is_vpn, is_proxy, is_tor
    (works without API key for 100 requests per day; API key is free)
-  https://ipqualityscore.com API - returns fraud score assessment of public IP address
    (requires API key - free for 5000 queries per month)
- post to https://api.ipblacklist.ai/checkIP with API key and IP address - provides info about blacklisting of provided IP
    (requires API key - free with registration, allows 1,000 daily requests)
- call to https://metrics.torproject.org/onionoo.html to check if IP address is listed in Tor directory
- call to Hetrixtools.com - or any other site that allows nmap port scanning from the outside
    (checking ICMP and any ports that look open)

A plugin page would let user enable/disable individual checks above (and provide API keys), would show the log of past checks.
A widget on the dashboard would show the lastest results

**Comment by mimugmail** [comment]
Nice idea! I'm in :)

**Comment by mihakralj** [comment]
Some progress here; Looking for ideas how to package this information to preserve real estate and add visual pizzaz... :-)
![image](https://user-images.githubusercontent.com/31756078/113808233-70a90100-971a-11eb-9450-cbbc87f486e8.png)

**Comment by mihakralj** [comment]
I have this prototype working and it takes about 10 seconds to load all tests when the widget loads - and I don't like that.
Questions:
- should the widget display results from the latest check performed - and not perform tests in real-time?
- How should we treat conflicting results? (one test returns true for VPN check and all the others return false)

**Reply by mimugmail** [reply]
If it takes 10 seconds it's not qualified to be a widget, better put the page in reporting like Speedtest and as a widget show the last results but with a note when they were taken?
